src/interpreter/vm.rs

```diff
@@ -335,16 +466,33 @@ where
                         ret.msg_receipt().exit_code()
                     );
                 }
+
+                if enable_tracing {
```

Comment:
I wonder if we could reduce the number of these branches in the code, e.g, have a dynamic object.

Reply:
#3405